### ISSUE TITLE
Reorder reaction buttons in reaction bar

### ISSUE THREAD

**Issue by AB-Y** [nexfoundation/vanilla#174]

**Comment by AB-Y** [comment]
**在文章 detail page 中，順序為**
Like, Comment, Bookmark, Share to FB

如圖中綠色框框內
<img width="1016" alt="截圖 2021-04-5 下午4 42 54" src="https://user-images.githubusercontent.com/38776510/114841756-c82a2980-9e0a-11eb-9b33-6f5410df2afe.png">

**在 comment 中，的順序為**
引用、Like、Share
如上圖的橘色框內

----
QA team 需檢查文章列表與單一文章頁面順序都是上述排序

**Comment by blueworrybear** [comment]
@kun-yu-tsai @AB-Y 

This request would be a hard one without modifying original source codes.
It's because that the **reactions** and **buttons* are rendered after the discussion controller firing an **beforeRender** event.
There are several **Plugins** (or **Pluggable**) will listen to the event, and attach the button to the webpage one by one.
Currently, it seems to me that there is no way to order how plugins handle the event, therefore it's hard to arrange the order of buttons.

Of course, we are able to change vanilla's source codes, or overwrite the whole controller, but it takes time to finish it.

The other possible solution would be hiding the original buttons and build to new plugin to render all required button at once.

---

In summary, there are **3 ways** to solve this issue as below:

1. Modifying original source code (*not recommended*)
2. Overwrite the discussion controller in the theme plugin (*lots of boilerplates*)
3. Hiding old buttons, using new plugin to render them at once. (*need diving into the REST API*)
